# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/CookieServiceChild.cpp
# Commit: d90c8b4228bd
# Full Hash: d90c8b4228bde30e5d3ced226975b5f353b68e63
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2020-07-09 21:37:35
# Description:
#   Bug 1650656 - Call CookieCommons::SchemeToSchemeType only when the scheme is supported, r=mayhemer
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D82491
# ==============================================================================

diff -r ec880b46ef45 -r d90c8b4228bd netwerk/cookie/CookieServiceChild.cpp
--- a/netwerk/cookie/CookieServiceChild.cpp	Thu Jul 09 12:40:43 2020 +0000
+++ b/netwerk/cookie/CookieServiceChild.cpp	Thu Jul 09 11:05:58 2020 +0000
@@ -331,6 +331,10 @@
 
   nsCOMPtr<nsIPrincipal> principal = aDocument->EffectiveStoragePrincipal();
 
+  if (!CookieCommons::IsSchemeSupported(principal)) {
+    return NS_OK;
+  }
+
   nsICookie::schemeType schemeType =
       CookieCommons::PrincipalToSchemeType(principal);
 
